cpp/cmake_modules/SetupCxxFlags.cmake

```diff
@@ -19,16 +19,30 @@
 # instruction sets that would boost performance.
 include(CheckCXXCompilerFlag)
 # x86/amd64 compiler flags
-check_cxx_compiler_flag("-msse4.2" CXX_SUPPORTS_SSE4_2)
-check_cxx_compiler_flag("-mavx2" CXX_SUPPORTS_AVX2)
-check_cxx_compiler_flag("-mavx512f" CXX_SUPPORTS_AVX512)
+if(MSVC)
+  set(ARROW_SSE4_2_FLAG "")
+  set(ARROW_AVX2_FLAG "/arch:AVX2")
+  set(ARROW_AVX512_FLAG "/arch:AVX512")
+  set(CXX_SUPPORTS_SSE2 TRUE)
+  set(CXX_SUPPORTS_SSE4_2 TRUE)
+else()
+  set(ARROW_SSE2_FLAG "-msse2")
+  set(ARROW_SSE4_2_FLAG "-msse4.2")
+  set(ARROW_AVX2_FLAG "-mavx2")
+  # skylake-avx512 consists of AVX512F,AVX512BW,AVX512VL,AVX512CD,AVX512DQ
+  set(ARROW_AVX512_FLAG "-march=skylake-avx512")
+  check_cxx_compiler_flag(${ARROW_SSE2_FLAG} CXX_SUPPORTS_SSE2)
+  check_cxx_compiler_flag(${ARROW_SSE4_2_FLAG} CXX_SUPPORTS_SSE4_2)
+endif()
+check_cxx_compiler_flag(${ARROW_AVX2_FLAG} CXX_SUPPORTS_AVX2)
+check_cxx_compiler_flag(${ARROW_AVX512_FLAG} CXX_SUPPORTS_AVX512)
 # power compiler flags
-check_cxx_compiler_flag("-maltivec" CXX_SUPPORTS_ALTIVEC)
+set(ARROW_ALTIVEC_FLAG "-maltivec")
```

Comment:
I don't think we should check any `-m` flag under MSVC. It will fail and takes a bit of time.

Reply:
(by a bit of time, I mean each of these checks seem to take 1 second on my Windows VM)

Reply:
Yes, better to avoid these unnecessary checking. Will do.
CMAKE_SYSTEM_PROCESSOR may break cross building, but I don't think it's reasonable to cross build a complex application like Arrow.